src/main/scala/com/fulcrumgenomics/util/Sequences.scala

```diff
@@ -150,15 +152,74 @@ object Sequences {
   }
 
   /** Reverse complements a string of bases. */
-  def revcomp(s: String): String = SequenceUtil.reverseComplement(s)
+  def revcomp(s: String): String = {
+    val bs = s.getBytes
+    revcomp(bs)
+    new String(bs)
+  }
 
   /** Returns the sequence that is the complement of the provided sequence. */
   def complement(s: String): String = {
     val bs = s.getBytes
-    forloop(from=0, until=bs.length) { i =>  bs(i) = SequenceUtil.complement(bs(i)) }
+    complement(bs)
     new String(bs)
   }
 
+  /** Reverse complements an array of bases in place. See [[complement()]] for how bases are complemented. */
+  def revcomp(bs: Array[Byte]): Unit = {
+    SequenceUtil.reverse(bs, 0, bs.length)
+    complement(bs)
+  }
+
+  /**
+    * Complements the bases in the array in place. `U` and `u` are always complemented to `A` and `a` respectively.
+    * IUPAC ambiguity codes are handled appropriately.  Any byte that is not a valid upper- or lower-case DNA base,
+    * RNA base or IUPAC code will cause an exception to be thrown.
+    *
+    * @param bs
+    */
+  def complement(bs: Array[Byte]): Unit = {
+    forloop (from=0, until=bs.length) { i =>
+      val complement: Byte = (bs(i): @switch) match {
+        case 'A' => 'T'
+        case 'C' => 'G'
+        case 'G' => 'C'
+        case 'T' => 'A'
+        case 'U' => 'A'
+        case 'M' => 'K'
+        case 'R' => 'Y'
+        case 'W' => 'S'
+        case 'S' => 'W'
+        case 'Y' => 'R'
+        case 'K' => 'M'
+        case 'V' => 'B'
+        case 'H' => 'D'
+        case 'D' => 'H'
+        case 'B' => 'V'
```

Comment:
Same, B->V and V->B, just have them one after the other.  Helps check correctness.